### DISCUSSION TITLE
Is bed level visualizer meant to be used with or without a steel sheet in place?

### DISCUSSION THREAD

**Discussion by bdlabitt** [jneilliii/OctoPrint-BedLevelVisualizer#644] [Q&A]
There's no mention of this anywhere that I can find.  Is this tool meant to be used with the steel sheet in place, or without?  If with, that means one has to wait for thermal equilibrium each time the test is performed?  For instance, I'd think that one would have to wait for both the hot end and heated bed to get hot - and wait a while for the structure to get warm as well?

I'd like to level the bed say for PETG.  First layer settings are 240C for hot end, and 85C for heated bed.  I'd imagine things are still expanding due to heat, since the heat has not flowed to the jacking screws or frame, even though the hot end and heat bed have reached temperature yet.

**Comment by jneilliii** [comment]
This is more of a general question relative to bed leveling as a whole not just within the plugin. Arguments go both ways on this one, but in my opinion having the bed at temp when leveling is the way to go. Pretty sure all the example scripts also have those temps being set. This is one of the benefits of UBL, is you can have different leveling meshes based on a bed temp to counter any changes due to that.

**Reply by bdlabitt** [reply]
Yes, I suppose you are correct.  In your experience, should the build plate (in my case a steel coated one) be on the heat bed?  It sure adds to the cycle time.  The reason I am asking is I am getting variation in the errors from scan to scan, and I am trying to narrow down the cause.  At this point just looking for ways to do this in an efficient manner.  So I was wondering if there was something that was a reasonable recipe for success.  As of the moment, I haven't converged on a flat enough bed, which begs the question, how flat is good enough?  Less than 0.05 mm error?  Less?

For a known good working system, what is the expected variation from scan to scan of a cell?  Like I said, I have issues with my print bed, so I'm trying to understand what's reasonable.  Haven't seen sources that describe "the standard deviation" of the measurement.  Do you have an estimate of that variation?  Not asking you to measure it, just trying to get some reasonable estimates based on your development of this visualizer.  Thanks.

**Reply by jneilliii** [reply]
I would have the build plate on the bed, as that can react differently than the bed itself to the temperature. In most cases what I tell people is if you can get the variance within a layer height (typically .2) then you are good to go. The whole point of having the bed leveling in the first place is to account for the variations, but you want to minimize it as much as possible so the printer's not having to move too much.

**Reply by bdlabitt** [reply]
Perhaps I am misunderstanding something.  Variation of a whole layer height (0.2mm) is acceptable?  Did you mean 0.02mm? Wouldn't a variation of one layer height over the build plate be awful?  Like total lack of adhesion in some places?  Please clarify.  Thank you.

**Reply by jneilliii** [reply]
> Variation of a whole layer height (0.2mm) is acceptable?

That's correct, a total variance from the lowest point to the highest point of **0.2mm** is usually good enough for auto bed leveling to do it's job without having to overwork the firmware. 

> Like total lack of adhesion in some places?

No, not if your z is moving like it's supposed to when going in the x and y direction while printing. This is the whole point of Auto Bed Leveling, the physical printing height of the layer should stay at exactly whatever you configure it in your slicer **relative** to the probed mesh.
